iaso/api/tasks/__init__.py

```diff
@@ -72,7 +72,7 @@ class TaskSourceViewSet(ModelViewSet):
     PATCH /api/tasks/<id>
     """
 
-    permission_classes = [permissions.IsAuthenticated, HasPermission(permission.DATA_TASKS)]  # type: ignore
+    permission_classes = [permissions.IsAuthenticated, HasPermission(permission.DATA_TASKS, permission.MOBILE_APP_OFFLINE_SETUP)]  # type: ignore
```

Comment:
This was needed to enable me to display a progress bar on the zip creation (which polled the `GET /api/tasks/:id` that was just created).

However, you're right that it is not clear, and too permissive, so I've changed the behavior of only the `GET /api/tasks/<id>` endpoint.

Basically: keep the tasks endpoint permissions as is, but allow a user to query the state of a task launched by themselves, even if they don't have the `DATA_TASKS` permission.

IMO it makes sense to allow this, but only on the `retrieve` endpoint and only for tasks created by the user themself.

@hakifran what do you think?

Reply:
@bramj I do think that the user who launch the task should have the  DATA_TASKS, if not don't let him launch the mobile app offline setup. Because on my point of view this can lead to a confusion about permissions. @beygorghor or @madewulf can give their point of view on this. thank you